torch/_inductor/compile_fx.py

```diff
@@ -420,13 +420,18 @@ def get_patched_config_dict(config_patches=None) -> Dict[str, Any]:
         return config.get_config_copy()
 
 
-@functools.wraps
```

Comment:
what was wrong with this?

Reply:
I dont really understand why you need to take arguments, instead of just function pointer here

Reply:
also you probably want to keep wraps around so that stacktraces do not change

Reply:
> what was wrong with this?

The previous implementation is checking `force_disable_caches` once when it evaluates the decoration, but then just returns the original function `compile_fx_inner`, which is executed directly thereafter. What we want is to return a function does the fresh_cache and then calls the decorated function.

For example, this test program:

```
disable = True

def decorator(f):
    if disable:
        print("Disabled")
        return f
    else:
        print("Enabled")
        return f

@decorator
def hello():
    print("hello world")

hello()
disable = False
hello()
```

Prints this output:
```
Disabled
hello world
hello world
```

But yeah, I accidentally removed the functools.wraps